cmd/server/handlers/handlers.go

```diff
@@ -0,0 +1,107 @@
+package handlers
+
+import (
+	"evgen3000/go-musthave-metrics-tpl.git/cmd/server/storage"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/go-chi/chi/v5"
+)
+
+type Handler struct {}
+
+func NewHandler() *Handler {
+	return &Handler{}
+}
+
+func (h Handler) HomeHandler(storage *storage.MemStorage) http.HandlerFunc {
+	return func(rw http.ResponseWriter, r *http.Request) {
+		body := "<h4>Gauges</h4>"
+		for gaugeName, value := range storage.GetAllGauges() {
+			body += gaugeName + ": " + strconv.FormatFloat(value, 'f', -1, 64) + "</br>"
+		}
+		body += "<h4>Counters</h4>"
+
+		for counterName, value := range storage.GetAllCounters() {
+			body += counterName + ": " + strconv.FormatInt(value, 10) + "</br>"
+		}
+		rw.Header().Set("Content-Type", "text/html; charset=utf-8")
+
+		_, err := rw.Write([]byte(body))
+		if err != nil {
+			log.Printf("Write failed: %v", err)
+		}
+	}
+}
+
+func (h Handler) UpdateMetricHandler(storage *storage.MemStorage) http.HandlerFunc {
+	return func(rw http.ResponseWriter, r *http.Request) {
+		metricType := chi.URLParam(r, "metricType")
+		metricName := chi.URLParam(r, "metricName")
+		metricValue := chi.URLParam(r, "metricValue")
+
+		switch metricType {
+		case "counter":
+			value, err := strconv.ParseInt(metricValue, 10, 64)
+			if err != nil {
+				http.Error(rw, "Bad request", http.StatusBadRequest)
+				return
+			}
+			storage.IncrementCounter(metricName, value)
+		case "gauge":
```

Comment:
[CHANGE REQUEST] This is a magic string.
https://clang.llvm.org/extra/clang-tidy/checks/readability/magic-numbers.html